### DISCUSSION TITLE
Custom trash/remove behavior

### DISCUSSION THREAD

**Discussion by fmorroni** [sxyazi/yazi#692] [1. Q&A] [answered]
Hi, is it possible to customize the behavior of the remove command? I already use `trashy` for deleting files, which has it's own trash directory and I don't want to have two separate ones.

Weirdly though, in my regular file system it already seems to be working as I want it to. ~No `.Trash` folder is created in my home~ and the deleted files go to the `trashy` trash directory. But on external drives like pendrives the `.Trash` is created in the mounted fs and it's annoying. ***Edit: Nvm in the main fs the `.Trash-1000` folder gets created in `/tmp`***

I guess it should be posible to just map `d` to `trashy put`, but I tried doing it like this:
`{ on = [ "d" ], exec = [ "escape --visual", "trashy put" ], desc = "Move the files to the trash" }`
and it's not doing anything. I'm probably doing it wrong but that's what I understood from the documentation...
Any help/tips is appreciated. Thanks!

**Comment by sxyazi** [comment, marked as answer]
Does

```rust
{ on = "d", run = [ "escape --visual", "shell -- trashy put %s" ] }
```

work for you? The `shell` command docs https://yazi-rs.github.io/docs/configuration/keymap/#mgr.shell

**Reply by fmorroni** [reply]
Ah I see how it's supposed to work now, that works thanks. But it seems that trashy actually uses the same convention than yazi which I thought wasn't the case, so it's also creating the `.Trash` folder for mounted drives after all haha
I guess I'll just stick with the default `remove` command then.
Thanks for the help and sorry for the trouble!

**Reply by sxyazi** [reply]
This might be because both Yazi and `trashy` adhere to the same [The FreeDesktop.org Trash specification](https://specifications.freedesktop.org/trash-spec/trashspec-1.0.html), and creating `.Trash` is part of this specification.
